### DISCUSSION TITLE
Selecting tracks via mouse drag in Playlist/Library

### DISCUSSION THREAD

**Discussion by TT-ReBORN** [TT-ReBORN/Georgia-ReBORN#113] [Ideas]
[takaji10](https://github.com/takaji10)
Do you know if it's possible to enable click and drag functionality in the playlist to select tracks? I'm wondering if SMP has this functionality; I think I saw it somewhere once but forget where...

---

[TT-ReBORN](https://github.com/TT-ReBORN)
What do you mean by click and drag?

You can select individual tracks via CTRL + click, when clicking again it will deselect.
SHIFT ( holding ) + click for example track 01 and then track 20 will select all in between.
Clicking on the album thumbnail will select the whole album.
Then mouse drag up/down to reorder the tracks.

---

[takaji10](https://github.com/takaji10)
Yes, I know about those :)

I meant like this: 
![image](https://user-images.githubusercontent.com/1534470/234778585-4d7d95b3-3a2f-47b6-bb95-554cb7ecf7d0.png)

---

[takaji10](https://github.com/takaji10)
Ah, I was thinking of SimPlaylist, which has this functionality. Although it hasn't been updated since 2011 :P

---

[TT-ReBORN](https://github.com/TT-ReBORN)
Ah ok, I do now understand what you mean, you want to select the items via mouse drag.

---

[takaji10](https://github.com/takaji10)
Yep, exactly - but if the functionality doesn't exist then I just need to stick to the ctrl+a. :)

**Comment by TT-ReBORN** [comment]
@takaji10, right know when doing mouse drag, it will automatically drag and drop items in the Playlist or in the Library.
You already know but I'm mentioning this again, to illustrate that these drag methods need to be enhanced, i.e code needs to be refactored/rewritten. I did not have time to investigate this yet...

-TT

**Reply by takaji10** [reply]
There's also some issue with the draw when using drag and drop onto the playlist - maybe you know:


https://github.com/TT-ReBORN/Georgia-ReBORN/assets/1534470/f7159589-2eb9-4c4a-a07d-4ad23541b1a1

Take your time on it!
